File: website/src/scripts/getDocId.test.js
import { getDocId } from './getDocId.js'; // Assuming getDocId.js is in the same directory
import fs from 'fs';
import { jest } from '@jest/globals'; // Import jest for spyOn
import path from 'path';
import { fileURLToPath } from 'url';

// Helper to get current directory in ES modules
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Load common test cases from JSON file
function loadCommonTestCases() {
    // Construct path relative to this test file
    // This file is in 'website/src/scripts/', shared_test_cases.json is in 'tests/'
    // So, ../../../tests/shared_test_cases.json
    const jsonPath = path.resolve(__dirname, '..', '..', '..', 'tests', 'get_doc_id_test_cases.json');
    const fileContent = fs.readFileSync(jsonPath, 'utf8');
    return JSON.parse(fileContent);
}

const COMMON_TEST_CASES = loadCommonTestCases();

describe('getDocId JavaScript', () => {
    test.each(COMMON_TEST_CASES)('getDocId(%s) should return %s', (docUrl, expectedId) => {
        // Suppress console.error for expected invalid URL parsing warnings during test
        const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
        expect(getDocId(docUrl)).toBe(expectedId);
        consoleErrorSpy.mockRestore();
    });
});

File: analyzer/shared/get_doc_id_test.py
import pytest
import json
import os
from get_doc_id import get_doc_id

# Load common test cases from JSON file
def load_common_test_cases():
    # Construct path relative to this test file
    # This file is in 'analyzer/', shared_test_cases.json is in 'tests/'
    # So, ../tests/shared_test_cases.json
    base_dir = os.path.dirname(os.path.abspath(__file__))
    json_path = os.path.join(base_dir, '..', 'tests', 'get_doc_id_test_cases.json')
    with open(json_path, 'r') as f:
        return [tuple(item) for item in json.load(f)]

COMMON_TEST_CASES = load_common_test_cases()

@pytest.mark.parametrize("doc_url, expected_id", COMMON_TEST_CASES)
def test_get_doc_id_specific_urls_python(doc_url, expected_id):
    print(f"Testing URL: {doc_url} with expected ID: {expected_id}")
    assert get_doc_id(doc_url) == expected_id

